src/arns.lua

```diff
@@ -700,12 +703,14 @@ end
 --- @param intendedAction IntendedAction The intended action to get token cost for
 --- @return TokenCostResult tokenCostResult The token cost result of the intended action
 function arns.getTokenCost(intendedAction)
+	local intent = intendedAction.intent
 	local tokenCost = 0
 	local purchaseType = intendedAction.purchaseType or "lease"
 	local years = tonumber(intendedAction.years)
 	local name = intendedAction.name
-	local baseFee = demand.baseFeeForNameLength(#name)
-	local intent = intendedAction.intent
+	local baseFee = demand.baseFeeForNameLength(
```

Comment:
you can see on line 761 we have this value set, not necessary to add this conditional here

Reply:
We need to add it because theres an assert in baseFeeForNameLength that it retrieved a fee - if we have a 51 char base name + an undername, this will attempt to access an out of bounds fee (eg length of 52 +) and throw an error.

One option was to use the getBaseName util, this seemed better at the time. Thoughts?

Reply:
yeah, lets use the getBaseName util and add a comment why it's necessary

Reply:
[e667966](https://github.com/ar-io/ar-io-network-process/pull/392/commits/e66796656ad100448e4db4f7aabe0678b6db5e9b)